src/components/Context/Context.tsx

```diff
@@ -0,0 +1,88 @@
+import React, {
+  createContext,
+  ReactNode,
+  useEffect,
+  useState,
+} from 'react';
+import { getProducts } from '../../api/products';
+import { ContextValue } from '../../types/ContextValue';
+import { Product } from '../../types/Product';
+import { Error } from '../../types/Error';
+import { ProductDetailsType } from '../../types/ProductDetails';
+
+export const Context = createContext<ContextValue>({
+  products: [],
+  selectedProduct: JSON.parse(`${localStorage.getItem('selectedProduct')}`),
+  cart: [],
+  favorite: [],
+  error: null,
+  isLoading: false,
+  setProducts: () => {},
```

Comment:
```suggestion
  setProducts(): void,
```
or better

```suggestion
  setProducts: Dispatch<SetStateAction<Product[]>>,
```

Reply:
the same everywhere

Reply:
I've set the following details in the type "ContextValue.ts".
In Context.tsx the mentioned values has been set as default values, not types.
```
export type ContextValue = {
  products: Product[],
  selectedProduct: ProductDetailsType,
  cart: Product[],
  favorite: Product[],
  error: Error | null,
  isLoading: boolean;
  setProducts: Dispatch<SetStateAction<Product[]>>,
  setSelectedProduct: Dispatch<SetStateAction<ProductDetailsType>>,
  setCart: Dispatch<SetStateAction<Product[]>>,
  setFavorite: Dispatch<SetStateAction<Product[]>>,
  setError: Dispatch<SetStateAction<Error | null>>,
  setIsLoading: Dispatch<SetStateAction<boolean>>,
};

```

Reply:
oh, my bad, sorry
I don't know how I didn't pay attention that its default values 🙄